mayastor/src/grpc/v1/pool.rs

```diff
@@ -0,0 +1,324 @@
+    async fn create_pool(
+        &self,
+        request: Request<CreatePoolRequest>,
+    ) -> GrpcResult<Pool> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                match PoolBackend::try_from(args.pooltype)? {
+                    PoolBackend::Lvs => {
+                        let rx = rpc_submit::<_, _, LvsError>(async move {
+                            let pool = Lvs::create_or_import(
+                                PoolArgs::try_from(args)?,
+                            )
+                            .await?;
+                            Ok(Pool::from(pool))
+                        })?;
+
+                        rx.await
+                            .map_err(|_| Status::cancelled("cancelled"))?
+                            .map_err(Status::from)
+                            .map(Response::new)
+                    }
+                }
+            },
+        )
+        .await
+    }
+
+    #[named]
+    async fn destroy_pool(
+        &self,
+        request: Request<DestroyPoolRequest>,
+    ) -> GrpcResult<()> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                info!("{:?}", args);
+                let rx = rpc_submit::<_, _, LvsError>(async move {
+                    if let Some(pool) = Lvs::lookup(&args.name) {
+                        if args.uuid.is_some() && args.uuid != Some(pool.uuid())
+                        {
+                            return Err(LvsError::Invalid {
+                                source: Errno::EINVAL,
+                                msg: format!(
+                                    "invalid uuid {}, found pool with uuid {}",
+                                    args.uuid.unwrap(),
+                                    pool.uuid(),
+                                ),
+                            });
+                        }
+                        pool.destroy().await?;
+                    } else {
+                        return Err(LvsError::Invalid {
+                            source: Errno::EINVAL,
+                            msg: format!("pool {} not found", args.name,),
+                        });
+                    }
+                    Ok(())
+                })?;
+
+                rx.await
+                    .map_err(|_| Status::cancelled("cancelled"))?
+                    .map_err(Status::from)
+                    .map(Response::new)
+            },
+        )
+        .await
+    }
+
+    #[named]
+    async fn export_pool(
+        &self,
+        request: Request<ExportPoolRequest>,
+    ) -> GrpcResult<()> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                info!("{:?}", args);
```

Comment:
so we have infos on enter, is there somewhere an info or other log with the result of the call?

Reply:
Yes there is a log in the destroy function called in this. I will try to make the logs consistent for all different RPCs. Preferably moving the logs from grpc file to lvs lib with proper sentences instead of just dumping the logs.

Reply:
hmm potential problem with moving it to the lvs, is that if a pool could not be found for example, we would not get the log, unless we have a specific log for the not found case...
but where will the log of the result go?

Reply:
I think it may be worth making use of tracing here because I can give us things like automatic logs on error etc.

Reply:
Added tracing to all the lvs functions called in the rpc function

Reply:
any reason to adding it to the lvs instead of the grpc? In the grpc we'd miss some errors, eg the ret error on 241.

Reply:
It shows error like this:
```
[2021-12-03T12:33:52.484652812+05:30 ERROR mayastor::grpc::v1::pool:pool.rs:232] error=status: InvalidArgument, message: "errno EINVAL: Invalid argument: pool tpool not found", details: [], metadata: MetadataMap { headers: {} }
```

As this will be returned as the grpc status anyways and no actual mayastor gets triggered I thought we can ignore such errors.

Reply:
hmm yeah good point, don't know... one pro to keeping it is that we have the info! message on start, so might look confusing if there is no corresponding end message.

Reply:
I checked all the called functions have info! message at the end in them and they look correct. I think adding an exit message will make it repetitive. Another issue I faced was once we pass the args then at the end we get the `already moved` error which again will require hacks like cloning the fields etc.